pkg/spi-shared/config/defaults.go

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"golang.org/x/oauth2"
+	"golang.org/x/oauth2/github"
+)
+
+// ServiceProviderDefaults configuration containing default values used to initialize supported service providers
+type ServiceProviderDefaults struct {
+	SpType   ServiceProviderType
+	Endpoint oauth2.Endpoint
+	UrlHost  string // default host of service provider. ex.: `github.com`
+	BaseUrl  string // default base url of service provider, typically scheme+host. ex: `https://github.com`
+}
+
+// all servise provider types we support, including default values
+var SupportedServiceProvidersDefaults []ServiceProviderDefaults = []ServiceProviderDefaults{
```

Comment:
If we used `ServiceProviderType` as the holder of the defaults, this could be called `SupportedServiceProviderTypes` which IMHO reads ever so slightly better.

Also, I would actually make the individual default or type instances public (and only reference them in this array) so that we could avoid the naming explosion for the constants specifying the default values of each of the fields.

E.g. instead of `config.GithubSaasHost` and `config.QuaySaasHost` and `config.GitlabSaasHost`, we'd have `config.GithubServiceProviderDefaults.SaasHost`, `config.QuayServiceProviderDefaults.SaasHost` and `config.GitlabServiceProviderDefaults.SaasHost`. IMHO, that is slightly more consistent and provides less room for typos and overall cognitive overload.